core/amber/src/main/scala/edu/uci/ics/texera/web/resource/CollaborationResource.java

```diff
@@ -0,0 +1,42 @@
+package edu.uci.ics.texera.web.resource;
+
+import edu.uci.ics.texera.web.ServletAwareConfigurator;
+
+import javax.websocket.*;
+import javax.websocket.server.ServerEndpoint;
+import java.util.HashMap;
+
+
+@ServerEndpoint(
+        value = "/collaboration",
```

Comment:
if the endpoint is a web socket endpoint, can you give it the prefix `/wsapi/`?

Reply:
do we necessarily need a separate websocket?

Reply:
@zuozhiw thinks it's better this way.

Reply:
also shall we use abbreviation like `collab`?

Reply:
for all the ws endpoints we need the prefix, because upon deployment we have some proxy mappings.  I don't know why it's better this way. also which way are you talking about?

Reply:
Using a separate websocket.

Reply:
@Xiao-zhen-Liu please add a `/wsapi` prefix as Yicong suggested because it's related to our production server proxy configuration, I forgot about this

As for a separate websocket, I feel it's better to make this module fully independent of our current functionailities, so let's use a separate websocket for now